dd-sdk-android/src/main/kotlin/com/datadog/android/rum/GlobalRum.kt

```diff
@@ -121,14 +135,30 @@ object GlobalRum {
 
     internal fun updateContext(rumContext: RumContext) {
         updateRumContext {
+            lastUserInteractionNs.set(System.nanoTime())
             rumContext.copy()
         }
     }
 
     internal fun getRumContext(): RumContext {
+        updateSessionIdIfNeeded()
         return activeContext.get()
     }
 
+    private fun updateSessionIdIfNeeded() {
```

Comment:
I'm wondering if this should not be synchronised. Otherwise, this might create very short sessions in some edge scenarios, no?

Reply:
indeed